server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTargetService.java

```diff
@@ -529,50 +547,59 @@ private void forceReplication(ForceSyncRequest request, ActionListener<Transport
         if (indexShard == null || indexShard.getReplicationEngine().isEmpty()) {
             listener.onResponse(TransportResponse.Empty.INSTANCE);
         } else {
-            startReplication(indexShard, new SegmentReplicationTargetService.SegmentReplicationListener() {
-                @Override
-                public void onReplicationDone(SegmentReplicationState state) {
-                    try {
-                        logger.trace(
+            // We are skipping any validation for an incoming checkpoint, use the shard's latest checkpoint in the target.
+            startReplication(
+                indexShard,
+                indexShard.getLatestReplicationCheckpoint(),
```

Comment:
nothing in this method call's listener has changed other than passing the shard's latest replicationCheckpoint to as second param.